lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2315,9 +2334,6 @@ CompilerType TypeSystemSwiftTypeRef::GetChildCompilerTypeAtIndex(
           if (llvm::StringRef(AsMangledName(type))
                   .endswith("sSo18NSNotificationNameaD"))
             return GetTypeFromMangledTypename(ConstString("$sSo8NSStringCD"));
-          // FIXME: Private discriminators come out in a different format.
-          if (result.GetMangledTypeName().GetStringRef().count('$') > 1)
-            return fallback();
```

Comment:
Here's a thought. In this case, we could have two levels of "fallback" logic:

```
if TypeRef type contains '$'
    if AST has type
        return AST type
    else
        return TypeRef type
```

Reply:
I've had some second thoughts about this fallback logic. A possible deal breaker is that the AST could contain fewer children, and that would make the index invalid. In other words, a runtime index could return some other child in the AST. To proceed, we'd have to compare the AST child to see that it matches the runtime child, modulo the private identifier, but that style of comparison is in-exact which could cause rare bugs.

Reply:
Could we add a `if GetNumChildren(typeref) == GetNumChildren(ast)` condition to the fallback and error out otherwise?

Reply:
I added the fallback with a child count check in 3dad37a.